qiskit/pulse/instructions/instruction.py

```diff
@@ -95,7 +99,7 @@ def operands(self) -> Tuple:
     @property
     def channels(self) -> Tuple[Channel]:
         """Returns channels that this schedule uses."""
-        return self._channels
+        raise NotImplementedError
```

Comment:
Channels info has been stored also in the `self._operands` and we have been updating both though this is just an overhead. This PR removes instance variable `self._channels`, and now position of channels in `self._operands` is instruction dependent. Thus this becomes abstract.

Reply:
`self.duration` is defined in the same way. I think difference between them, i.e. `@abc.abstractproperty` and `raise NotImplementedError`, is if instantiation of the class is allowed or not. As your said, this property is called from constructor, so this seems to be abstract property since `Instruction` cannot be initialized without this.